lib/fund_america/api.rb

```diff
@@ -5,19 +5,15 @@ class << self
       # This method is called from each end point method to make API requests
       # using HTTParty gem. Takes the method, uri and options as input
       # Handles response and errors
-      def request method, uri, options={}
-        options = FundAmerica.basic_auth.merge!({:body => options})
+      def request(method, uri, options = {})
+        options = FundAmerica.basic_auth.merge!(body: options)
         uri = FundAmerica.base_uri + uri unless uri.include?('test_mode')
-        response = HTTParty.send(method, uri, options)
-        parsed_response = JSON.parse(response.body)
-        if response.code.to_i == 200
-          # Returns parsed_response - a hash of response body
-          # if response is successful
-          parsed_response
-        else
-          # Raises error if the response is not sucessful
-          raise FundAmerica::Error.new(parsed_response, response.code.to_i)
-        end
+        @response = HTTParty.send(method, uri, options)
+
+        raise_wrong_status_error unless success_response
+
+        parsed_response
+      rescue JSON::ParserError
```

Comment:
IMHO you should also catch other exceptions that can be thrown by HTTParty for example. In case of timeouts, connection issues, domain issues on FundAmerica side, etc. it can throw exceptions. I think BF should be able to catch those exceptions, to do that if would be good if gem could catch HTTParty exceptions and throw `FundAmerica::Error`
https://stackoverflow.com/a/26768085
Here example of solving such an issue by CC gem 
https://github.com/CurrencyCloud/currencycloud-ruby/blob/master/lib/currency_cloud/request_handler.rb#L45